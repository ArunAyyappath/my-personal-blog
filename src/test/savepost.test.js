import React from "react";
import SaveChangeButton from "../component/saveChanges";
import { configure, shallow, cleanup, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

afterEach = () => {
  cleanup;
};

describe("<SaveChangeButton/>", () => {
  it("renders correctly", () => {
    const wrapper = renderer.create(<SaveChangeButton />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it("verify whether the error throw is updated to state", () => {
    const wrapper = mount(<SaveChangeButton />);
    wrapper.setProps({ chekCancelStatus: true, titledata: "", backImg: "" });
    wrapper.find("button").simulate("click");
    expect(wrapper.instance().state).toStrictEqual({
      saveDataToDb: true,
      statusInSave: true,
      natureOfErrInSave: "error",
      messageInSave: "please fill Title & Upload Image",
      preloader: true
    });
  });
  it("verify whether the error throw is updated to state", () => {
    const wrapper = mount(<SaveChangeButton />);
    const spy = jest.spyOn(wrapper.instance(), "getUpdatedWithProps");
    wrapper.setProps({ chekCancelStatus: true, titledata: "", backImg: "" });
    wrapper.find("button").simulate("click");
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it("verify the update and cancel button have been rendered correctly", () => {
    const wrapper = mount(<SaveChangeButton />);
    wrapper.setProps({ chekCancelStatus: false });
    expect(wrapper.find("button").length).toBe(2);
  });

  it("verify the save button have been rendered correctly", () => {
    const wrapper = mount(<SaveChangeButton />);
    wrapper.setProps({ chekCancelStatus: true });
    expect(wrapper.find("button").length).toBe(1);
  });
});
