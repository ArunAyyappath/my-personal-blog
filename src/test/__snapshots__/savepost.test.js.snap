// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveChangeButton/> renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "90px 30px",
      "textAlign": "right",
    }
  }
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#2283c7",
        "color": "#fff",
        "margin": "5px",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
        />
      </svg>
      Save
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#2283c7",
        "color": "#fff",
        "margin": "5px",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Cancel
    </span>
  </button>
</div>
`;
